### PR TITLE
Add support for vapor 2

### PR DESCRIPTION
**Pull request by gperdomor**
This PR add support for Vapor 2 :D

This break the code. So should be tagged has version 2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,63 +1,128 @@
...
-	.Package(url: "https://github.com/SwiftyBeaver/SwiftyBeaver-Vapor.git", majorVersion: 1),
-	//...other packages here
-],
+import PackageDescription
+
+let package = Package(
+    name: "Project",
+    dependencies: [
+        .Package(url: "https://github.com/vapor/vapor.git", majorVersion: 2),
+        .Package(url: "https://github.com/SwiftyBeaver/SwiftyBeaver-Vapor.git", majorVersion: 2),
```
Comment (natebird) [inline README.md:31]: While I see you want to provide context for where this line should go I think you really only need what was originally there changed to version 2 of SwiftyBeaver-Vapor.

```
dependencies: [
	.Package(url: "https://github.com/SwiftyBeaver/SwiftyBeaver-Vapor.git", majorVersion: 2),		 
 	//...other packages here
 ],

Comment (gperdomor) [reply README.md:31]: @natebird I try to maintain the docs similar to official Vapor documentation, for example: http://beta.docs.vapor.codes/mysql/package/

Comment (natebird) [reply README.md:31]: Ah. I see. That's fine.

Comment (skreutzberger) [reply README.md:31]: Thanks! I think the whole installation instructions can stay since even not the major version will change.

**Comment by skreutzberger** [comment]
Looks great! You even added Travis! Please put the README changes in another PR and once done I am good to go to merge this PR 🥇

**Comment by skreutzberger** [comment]
oh wait, can you maybe also move the Travis stuff in an own PR because it fixes another issue? Thanks!

**Comment by gperdomor** [comment]
@skreutzberger just to clarify, remove the readme and ci changes? and send other PR?... The Installation and setup changed a little, the configuration is made using a `swiftybeaver.json` file so i think the readme is important for this PR

**Comment by skreutzberger** [comment]
@gperdomor ah ok, then please leave the README changes in and just keep the SPM setup instruction as they are (since they don’t change as far as I understand).

**Comment by gperdomor** [comment]
@skreutzberger removed CI files... Waiting for this to send the new PR with travis and circle CI files

**Comment by skreutzberger** [comment]
I tried your code and I get a linking error when running `swift test` under Ubuntu 14.04:

```
Linking ./.build/debug/SwiftyBeaverVaporExample
clang: error: unable to execute command: Killed
clang: error: linker command failed due to signal (use -v to see invocation)
<unknown>:0: error: link command failed with exit code 254 (use -v to see invocation)
```

I did the following in vanilla Ubuntu 14.04 on Digital Ocean as root user:
```
# install Swift 3.1:
wget https://swift.org/builds/swift-3.1-release/ubuntu1404/swift-3.1-RELEASE/swift-3.1-RELEASE-ubuntu14.04.tar.gz
tar zxf swift-3.1-RELEASE-ubuntu14.04.tar.gz
mv swift-3.1-RELEASE-ubuntu14.04 /swift-3.1
# ... added the PATH to ~/.bashrc
which swift   # shows /swift-3.1/usr/bin/swift
swift --version  # shows 3.1
curl -sL check.vapor.sh | bash

# then I cloned your code and built it
swift test 
```

**Comment by skreutzberger** [comment]
Maybe show that your code is working by providing a docker container. You can maybe get inspired by this one: https://github.com/SwiftyBeaver/SwiftyBeaver/blob/master/Dockerfile

**Comment by gperdomor** [comment]
@skreutzberger I decided use the new Vapor 2 template to make it works, for that reason basically this is a rewrite of all the code, by default Vapor 2 use 2 targets, `App` (which has the app code) and `Run` (executable, only for initialize and run the app), i add a new target `SwiftyBeaverProvider`... You need to run the `Run` executable, or `vapor run` command, SwiftyBeaverVaporExample not longer exist.

**Comment by gperdomor** [comment]
screen shot 2017-05-04 at 9 10 15 am

**Comment by skreutzberger** [comment]
I tried to install it using `vapor run` and it failed, too. Can you maybe add a Dockerfile which shows that is working? Thanks!

**Comment by gperdomor** [comment]
use the official vapor image... Enter to the project directory and run `docker run --rm -ti -v $(pwd):/vapor vapor/vapor:latest vapor build` and then `docker run -ti -v $(pwd):/vapor -p 8080:8080 vapor/vapor:latest vapor run`

screen shot 2017-05-04 at 10 13 38 am

Which version of vapor toolbox and swift you have?

**Comment by gperdomor** [comment]
@skreutzberger it works?

**Comment by agordeev** [comment]
@gperdomor how to try out your version? Not sure how to specify branch with Swift Package Manager.

**Comment by gperdomor** [comment]
@agordeev SPM works with tags, i prepared one in my repo, try this .Package(url: "https://github.com/gperdomor/SwiftyBeaver-Vapor.git", Version(2,0,0, prereleaseIdentifiers: ["beta"]))

**Comment by agordeev** [comment]
@gperdomor That worked, thank you!

**Comment by gperdomor** [comment]
@skreutzberger Vapor 2 is stable now, can we merge this?

**Comment by gperdomor** [comment]
A new provider has ben releases under vapor-community org...

**Comment by skreutzberger** [comment]
Thanks! I never made your solution work so I can not merge it into the official repository. But if it is working for you then it is a good idea to release it somewhere else. I will point to it once multiple sources confirm that it is stable or merge it in 👍

**Comment by agordeev** [comment]
@gperdomor too sad you had to delete your fork, it worked just fine for me. Where can I find it now?

**Comment by gperdomor** [comment]
@agordeev Hi, you can find it here https://github.com/vapor-community/swiftybeaver-provider

**Pull request closed without merging**